### PR TITLE
fix: call gui not visible on devices with android version > android 11

### LINKED ISSUE #499: Starting a voice call from Fluffychat Android doesn't open any GUI for the call
### Bug Description

When you start the call from the app, you have no way of interracting with it. All it does is sends a message stating "[Username] started a call". The only way to end the call is for the other person to hang up, and you close the app.

### Steps to Reproduce

1. Go to a chat with someone
2. Start a call 
3. observe that no GUI pops up

### Expected Behavior

I would expect some form of GUI to pop up to allow me to have basic interraction with the call

### App Version

1.13.0

### Additional Platform Information

Google Pixel 6

### Additional Context

This may be a feature request instead, but I'm assuming this functionality already exists, and isn't being triggered.

### PR DESCRIPTION
**Pull request by aryanA101a**
fixes #499 #500

### CODE REVIEW

**Comment by aryanA101a** [comment]
@krille-chan please have a look at it

File: lib/utils/voip/callkeep_manager.dart [hunk 1]
```diff
@@ -365,3 +385,8 @@ class CallKeepManager {
     setCallHeld(event.callUUID, event.hold);
   }
 }
+
+Future<PermissionStatus?> tryReadPhoneStatePermission() async {
+  final info = await DeviceInfoPlugin().androidInfo;
```
Comment (krille-chan) [inline lib/utils/voip/callkeep_manager.dart:399]: Doesn't this need a platform check?

File: lib/utils/voip/callkeep_manager.dart [hunk 2]
```diff
@@ -167,7 +168,26 @@ class CallKeepManager {
     Logs().v('[onPushKitToken] token => ${event.token}');
   }
 
-  Future<void> initialize() async {
+  Future<void> initialize({BuildContext? context}) async {
+    tryReadPhoneStatePermission().then((pStatus) {
+      if (context != null) {
+        if (pStatus == PermissionStatus.denied ||
+            pStatus == PermissionStatus.permanentlyDenied) {
+          ScaffoldMessenger.of(context).showSnackBar(
```
Comment (krille-chan) [inline lib/utils/voip/callkeep_manager.dart]: What happens if a user will permanently deny the permission? Will he gets a snackbar on any app start with the same content?

Comment (aryanA101a) [reply lib/utils/voip/callkeep_manager.dart]: Okay, I'll add a different message to the snackbar for permanently denied.
What do you prefer more? After getting permanently denied, show the snack bar only once or everytime the user opens the app?

File: lib/utils/voip/callkeep_manager.dart (hunk 1 above)
Comment (aryanA101a) [reply lib/utils/voip/callkeep_manager.dart:399]: Yes, I'll fix it

**Comment by aryanA101a** [comment]
@krille-chan I have made the changes, please have a look at it.

**Comment by mayemu** [comment]
This is still an issue, should be opened

**Pull request closed without merging**